### DISCUSSION TITLE
Minio replication status and delay

### DISCUSSION THREAD

**Discussion by gdm000** [minio/minio#13506] [Q&A] [answered]
I have two test clusters (actually 2 VMs with 4 drives aka folders for each VM) that're configured to be in Active-Active sync replication mode:

```
vm1: minio server minio/data/srv1/d{1..4} &
vm2: minio server minio/data/srv2/d{1..4} &

mc alias set mi1 <vm1> <creds>
mc alias set mi2 <vm2> <creds>

mc admin policy add mi1 ReplicationAdminPolicy minio/ReplicationAdminPolicy.json #(from minio docs)
mc admin user add mi1 mi1ReplicationAdmin <creds>
mc admin policy set mi1 ReplicationAdminPolicy user=mi1ReplicationAdmin

mc admin policy add mi2 ReplicationAdminPolicy minio/ReplicationAdminPolicy.json #(from minio docs)
mc admin user add mi2 mi2ReplicationAdmin <creds>
mc admin policy set mi2 ReplicationAdminPolicy user=mi2ReplicationAdmin

mc admin policy add mi1 ReplicationRemoteUserPolicy minio/ReplicationRemoteUserPolicy.json #(from minio docs)
mc admin user add mi1 mi1ReplicationRemoteUser <creds>
mc admin policy set mi1 ReplicationRemoteUserPolicy user=mi1ReplicationRemoteUser

mc admin policy add mi2 ReplicationRemoteUserPolicy minio/ReplicationRemoteUserPolicy.json #(from minio docs)
mc admin user add mi2 mi2ReplicationRemoteUser <creds>
mc admin policy set mi2 ReplicationRemoteUserPolicy user=mi2ReplicationRemoteUser

mc alias set mi1Repl <vm1> mi1ReplicationAdmin <creds>
mc alias set mi2Repl <vm2> mi2ReplicationAdmin <creds>

mc admin bucket remote add mi1Repl/test1 http://mi2ReplicationRemoteUser:<creds>@<vm2> --service "replication" --sync

Remote ARN1 = `arn:minio:replication::...:test1`.

mc admin bucket remote add mi2Repl/test1 http://mi1ReplicationRemoteUser:<creds>@<vm1>/test1 --service "replication" --sync

Remote ARN2 = `arn:minio:replication::...:test1`.

mc replicate add mi1Repl/test1 --remote-bucket test1 --arn '<Remote ARN1>' --replicate "delete,delete-marker,existing-objects"
mc replicate add mi2Repl/test1 --remote-bucket test1 --arn '<Remote ARN2>' --replicate "delete,delete-marker,existing-objects"
```
It works perfectly until one of the clusters is down. Second cluster (which is alive) continues to serve write requests and enqueues replication events. The problem is that it takes too much time to synchronize clusters when both or them're back online.
In my tests it takes about 15-20 mins to deliver missed objects from one cluster to another. It was just 10 files 2Gb size for bucket in total and it was no additional load on used VMs and network (new writes're processed at 60Mb/s speed) Files were appearing one by one with huge delay (several minutes for files with ~4kb size) 

So, I have several questions abount minio replication process:

Is it any way to adjust scraping process to do syncronization faster?

How to monitor the replicatation process to know that all files're successfully transferred to target replication bucket? 

Each file has a replication status, is it possible to list files that're still not replicated?

**Comment by harshavardhana** [comment, marked as answer]
Just set `mc admin config set alias scanner delay=0.0`

**Reply by klauspost** [reply]
There will always be a delay, since failed replications are picked up by the scanner, when it gets to it.

Thus order is not guaranteed and replication time is "eventually".

**Reply by gdm000** [reply]
> Just set `mc admin config set alias scanner delay=0.0`

Minio has this config but I can't find and documentation on it. Is there any place where I can find more details about this configuration?

**Reply by gdm000** [reply]
> There will always be a delay, since failed replications are picked up by the scanner, when it gets to it.
> 
> Thus order is not guaranteed and replication time is "eventually".

That's ok, but how can I see, which files're still not replicated? (if I need to wait for specific file on the target claster, whithout listening replication events, just list of files states on demand)

**Reply by klauspost** [reply]
https://docs.min.io/docs/minio-server-configuration-guide.html See "Usage scanner"
